src/borg/remote.py

```diff
@@ -755,8 +766,11 @@ def handle_error(unpacked):
                             raise UnexpectedRPCDataFormatFromServer(data)
                         if msgid in self.ignore_responses:
                             self.ignore_responses.remove(msgid)
+                            # async methods never return values, but may raise exceptions.
                             if b'exception_class' in unpacked:
-                                handle_error(unpacked)
```

Comment:
Good point, maybe this shouldn't be exposed at all externally, rather, have rollback *act* like a barrier. rollback could get a parameter such as "ignore_previous_errors" or so that discards errors from previous RPC calls. I have to think a bit if that's safe behaviour though.